### ISSUE TITLE
Marker Click Area Size

### ISSUE THREAD

**Issue by dnlsandiego** [istarkov/google-map-react#316]
I seem to have a problem expanding a map marker's click area size. Click events on markers only get triggered when clicking the very top left area of a marker. I tried following along the examples, specifically this one: http://istarkov.github.io/google-map-react/map/events/. This might be a usage issue but I can't seem to find anything in the docs for the API to remedy this. I tried to look through the issues and Stack Overflow but no luck. 

A modified version of the JSBin example to recreate the issue: https://jsbin.com/lumapiwule/1/edit?js,console,output

```JS
class SimpleMap extends React.Component {
  state = {
    center: [60.938043, 30.337157],
    zoom: 9,
  };

  _onChange = ({center, zoom}) => {
    this.setState({
      center: center,
      zoom: zoom,      
    });
  }

  render() {
    return (
       <GoogleMapReact
        onChange={this._onChange}
        onChildClick={(e) => console.log(e)}
        center={this.state.center}
        zoom={this.state.zoom}>
        <div className="place" lat={60.955413} lng={30.337844}>MyPlace</div>
      </GoogleMapReact>
    );
  }
}


ReactDOM.render(
  <div style={{width: '100%', height: 400}}>
    <SimpleMap/>
  </div>,
  document.getElementById('main')
);
```

how can I expand the click area of a map marker?

**Comment by istarkov** [comment]
Few days ago I've created a new doc. 
Even it's not finished your problem is described there, please read carefully https://github.com/istarkov/google-map-react/blob/master/DOC.md#few-markers-with-hover-example-done-right
I think that explanation and example will help.

**Comment by ulilicht** [comment]
Hello all! I have the same problem. I wonder if there is no simple way to solve this without creating a custom `distanceToMouse` function? 

I added a hover to the example of @dnlsandiego, you can find it here: https://jsbin.com/zuwexohowi/9/edit?js,console,output 

I don't understand why in default setup it doesn't work with rectangular markers?

**Comment by istarkov** [comment]
@ulilicht default setup is just for circle markers with hoverDistance eq to 30, I see no reason why it must be rectangle or has any other default form, circle is the simplest algorithmic form so I choose it. 
You are free to not use all `onChild*` event handlers, just use `onMouseEnter, Leave, onClick` event handlers on you markers. 

As like as here https://github.com/istarkov/google-map-react/blob/master/DOC.md#few-markers-with-hover-example

I think solution to hover problem I provide, works very well for all situations I have (and be sure I have enormous number of different controls with this map) 
I see no way to make it easier, as what can be easier than distance function.

**Comment by ulilicht** [comment]
Thank you very much @istarkov - I fixed my problem now. The main reason is for my problem was, that I didn't understand that the normally clickable area for onChildClick is always a circle. 

I solved my Problem with 3 steps: 
1) I added onClick handlers to my Markers and handling the click events myself
2) I corrected the distanceToMouse function by an offset so that it still used a circle but originates not around the lat/lng but around my marker which is on top of this. 
3) for styling purpose, I use `:hover` for highlighting the clickable area but add a class for `props.$hover` that pushes the `z-index` so that the correct marker is set on top. 

```
    distanceToMouse = ({ x, y }, { x: mouseX, y: mouseY }) => {
        const offsetY = MAP_MARKER_HEIGHT / 2;
        return Math.sqrt((x - mouseX) * (x - mouseX) + (y - mouseY - offsetY) * (y - mouseY - offsetY))
    }
```

Thanks for the great library, you just sometimes need to get used to using it ;-)
